ci-operator/step-registry/telco5g/jobs-router/telco5g-jobs-router-commands.sh

```diff
@@ -10,45 +10,81 @@ echo "************ telco cluster setup command ************"
+export DEBUG_T5CI_VERSION='${T5CI_VERSION}'
+export DEBUG_PROW_JOB_ID='${PROW_JOB_ID}'
+export DEBUG_JOB_NAME='${JOB_NAME}'
+export DEBUG_JOB_TYPE='${JOB_TYPE}'
+export DEBUG_RELEASE_IMAGE_LATEST='${RELEASE_IMAGE_LATEST-}'
+export GIT_COMMITTER_NAME='CI User'
+export GIT_COMMITTER_EMAIL='cnf-devel@redhat.com'
+#######################################################
+EOF
 
 if [[ "$PROW_JOB_ID" = *"nightly"* ]] && [[ "$JOB_TYPE" == "periodic" ]]; then
-    echo "export T5_JOB_TRIGGER=nightly" >> $MAINENV
-    echo "export T5_JOB_DESC=nightly-${T5CI_JOB_TYPE}" >> $MAINENV
-    echo "export T5_JOB_DESC_FULL=nightly-${T5CI_JOB_TYPE}-${T5CI_VERSION}" >> $MAINENV
     if [[ -n "${RELEASE_IMAGE_LATEST-}" ]]; then
         export IMG=${RELEASE_IMAGE_LATEST-}
     elif [[ -n "${RELEASE_IMAGE_INITIAL-}" ]]; then
         export IMG=${RELEASE_IMAGE_INITIAL-}
     else
         export IMG=${PROW_JOB_ID/-telco5g/}
     fi
-    echo "export T5_JOB_IMAGE=${IMG}" >> $MAINENV
-    # In case of running on nightly releases we need to figure out what release exactly to use
-    echo "export KCLI_PARAM=\"-P openshift_image=registry.ci.openshift.org/ocp/release:${IMG}\"" >> $MAINENV
-    echo "export SNO_PARAM=\"-f ${IMG}\"" >> $MAINENV
+    IMG_URL="registry.ci.openshift.org/ocp/release:$IMG"
+
+    cat << ____EOF >> "$MAINENV"
+export T5_JOB_TRIGGER=nightly
+export T5_JOB_DESC='nightly-${T5CI_JOB_TYPE}'
+export T5_JOB_DESC_FULL='nightly-${T5CI_JOB_TYPE}-${T5CI_VERSION}'
+# In case of running on nightly releases we need to figure out what release exactly to use
+export T5_JOB_RELEASE_IMAGE='$IMG_URL'
+export KCLI_PARAM='-P openshift_image=$IMG_URL'
+export SNO_PARAM='-f ${IMG}'
+____EOF
+
 elif [[ "$JOB_TYPE" == "periodic" ]]; then
-    echo "export T5_JOB_TRIGGER=periodic" >> $MAINENV
-    echo "export T5_JOB_DESC=periodic-${T5CI_JOB_TYPE}" >> $MAINENV
-    echo "export T5_JOB_DESC_FULL=periodic-${T5CI_JOB_TYPE}-${T5CI_VERSION}" >> $MAINENV
-    echo "export KCLI_PARAM=\"-P tag=${T5CI_VERSION} -P version=nightly\"" >> $MAINENV
-    echo "export SNO_PARAM=\"-t ${T5CI_VERSION} -r nightly\"" >> $MAINENV
+    # Identify release image url
+    # Ref.: https://github.com/karmab/kcli/blob/main/kvirt/cluster/openshift/disconnected/scripts/03_mirror.sh
+    if [ "$(echo "$T5CI_VERSION" | tr -dc '.' | wc -c)" -ge 2 ]; then
+        TAG="$T5CI_VERSION"
+    else
+        TAG="latest-$T5CI_VERSION"
+    fi
+    RELEASE_SUMMARY_URL="https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/$TAG/release.txt"
+    IMG_URL=$(curl -q -L -s "$RELEASE_SUMMARY_URL" | grep 'Pull From: quay.io' | awk -F ' ' '{print $3}')
+
+    cat << ____EOF >> "$MAINENV"
+export T5_JOB_TRIGGER=periodic
+export T5_JOB_DESC='periodic-${T5CI_JOB_TYPE}'
+export T5_JOB_DESC_FULL='periodic-${T5CI_JOB_TYPE}-${T5CI_VERSION}'
+export T5_JOB_RELEASE_IMAGE='$IMG_URL'
+export KCLI_PARAM='-P tag=${T5CI_VERSION} -P version=nightly -P openshift_image=$IMG_URL'
```

Comment:
No need to specify `openshift_image` in `kcli` when using `version` and `tag`, it calculates it. Anyway `openshift_image` won't be used in this case.
In general we don't need to identify the exact image here, because we just pass `latest` to `kcli` and that's it. But fine to have it, maybe we'll need it some day. Although it might be check for failures for curl and workaround case when no latest.

Reply:
The only reason i put `openshift_image` here is to ensure that kcli really uses the image that we calculated before. In case of a disconnected registry this is used but i see now that kcli does not apply it consistently, e.g. [it is only used when version is `ci` but not when version is `nightly`](https://github.com/karmab/kcli-openshift4-baremetal/blob/5352c7e9d008f670b94f957a303f3cd0899e4b2d/kcli_pre.sh#L72)

Reply:
Done.